src/checksum/arm64/crc32-arm64.c

```diff
@@ -51,18 +51,19 @@
     __asm__("crc32cb %w[c], %w[c], %w[v]" : [c] "+r"(*&crc) : [v] "r"(+value))
 
 /*
- * __wt_checksum_hw --
- *     Return a checksum for a chunk of memory, computed in hardware using 8 byte steps.
+ * __wt_checksum_with_seed_hw --
+ *     Return a checksum for a chunk of memory, computed in hardware using 8 byte steps. Start with
+ *     the given seed.
  */
 static uint32_t
-__wt_checksum_hw(const void *chunk, size_t len)
+__wt_checksum_with_seed_hw(uint32_t seed, const void *chunk, size_t len)
 {
     uint32_t crc;
     size_t nqwords;
     const uint8_t *p;
     const uint64_t *p64;
 
-    crc = 0xffffffff;
+    crc = ~seed;
```

Comment:
It is general practise not to use zero as seed value. I think it is better to through an error if the seed value given is zero. 
By negating we are changing '0' to 0xffffffff, but if the input seed value is 0xffffffff then the seed value '0' will be used.
My preference is to return error if the input seed value is '0'

Reply:
Here, the seed refers to the starting value. The algorithm for the CRC computation starts with an initial CRC of 0xffffffff, this is because, at the end, the computed CRC is flipped before returning to the user. The expectation is that for null/empty data the returned CRC will be 0, so hence starting with a seed of 0xffffffff.

Reply:
Google's cityhash also uses "withSeed" terminology:
https://github.com/google/cityhash/blob/f5dc54147fcce12cefd16548c8e760d68ac04226/README#L128

Linux kernel's crc32c just calls it "crc" for the starting argument.

Reply:
May be my comment is not clear.
I have no problem wiht seed or crc terminology.

My concern is : "consider a CRC initialized
with an all zero seed value. Any number of all zero dataword chunks can be fed into the CRC
and it will stay at a zero value, throwing away information about how many zero chunks have
been processed. However, if the CRC computation is initialized to any non-zero value,
processing chunks with a value of zero will cycle patterns through the CRC, tracking the number
of zero-valued chunks that have been processed as a sort of counter. This can help with
situations in which additional leading-zero bytes have been erroneously added to a dataword." from the [blogpost]( http://checksumcrc.blogspot.com/2019/08/effect-of-initial-seed-values-on-crcs.html)

My understanding of the above statement is the CRC value calculated will be same for the below two chunks, if the seed value or initial value is zero.
Chunk 1: "0x0000000000000000000000000000000000000000000000000000000000000abc"
Chunk 2: "0x000000abc"

So I think it is better to flag error if the input seed value could be '0'.
if the input seed value is 0xffffffff, then the initial value will be '0' and hence the crc generated for both the above chunks will be same. 

So my suggestion is to flag error if the input seed value given by the application can result in starting crc value as '0'.

If this comment is not clear, please let me know.

Reply:
Is there anything that prevents `0xffffffff` being returned as a legal checksum value? 
When we chain seeded function calls together we pass the output crc as a seed into the subsequent calls, so I can see the following happening (code snippet from `ex_all`):
```c
        crc32c = func(buffer, chunk1);
        // `func` returns a valid crc32 value of 0xffffffff. The next `func_with_seed` will reject it and return an error
        crc32c = func_with_seed(crc32c, buffer + chunk1, chunk2);
        crc32c = func_with_seed(crc32c, buffer + chunk1 + chunk2, chunk3);
```

With a 32 bit checksum this should be happening once every ~4 billion checksum calls which will be very common.

Reply:
Well, if all bits are set, for instance:
https://github.com/wiredtiger/wiredtiger/blob/e0db92acca87537c651fed72e730a2e9139007ae/test/unittest/tests/test_crc32.cpp#L28

So ```0xffffffff``` is technically valid. This was my argument against adding an assert. But this conversation has not yet concluded. Since I am more focused on getting zseries to behave correctly first. We will come back to this conversation.

Reply:
We concluded this discussion in a meeting. Now that I have added testing around buffers initialised to 0xFF, it has become clear that we will need to support a starting seed that could be set to that value. Closing this comment.